src/reconcile/rec_visibility.c

```diff
@@ -251,8 +258,9 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
             continue;
         }
 
-        if (upd->prepare_state == WT_PREPARE_LOCKED ||
-          upd->prepare_state == WT_PREPARE_INPROGRESS) {
+        /* Ignore prepared updates if it is not eviction. */
+        if (!F_ISSET(r, WT_REC_EVICT) && (upd->prepare_state == WT_PREPARE_LOCKED ||
```

Comment:
It is better to wait if the prepared state of an update is "WT_PREPARED_LOCKED" as this state is transient. Also since eviction runs in exclusive access, we should never encounter this case for eviction, we can add an assert to validate this assumption.

Reply:
Removed the handing scenarios of `WT_PREPARED_LOCKED` and added an assert to make sure that it doesn't occur.